File: leetcode/1137. 第 N 个泰波那契数/main.go
package main

import "fmt"

func main() {
	fmt.Println(tribonacci(4))
}

func tribonacci(n int) int {
	if n < 1 {
		return 0
	}
	if n <= 2 {
		return 1
	}
	if n <= 3 {
		return 2
	}
	return tribonacci(n-1) + tribonacci(n-2) + tribonacci(n-3)
}

File: leetcode/1137. 第 N 个泰波那契数/main.js
/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function(n) {
    total = 0
    curr = 1
    pre = 1
    prepre = 0
    if (n===0) {
        return 0
    }
    if (n===1) {
        return 1
    }
    if (n===2) {
        return 1
    }
    for (i = 3; i <= n; i++) {
        total = curr + pre +prepre
        prepre = pre
        pre = curr
        curr = total
    }
    return total
};